### PR TITLE
Ensure we don't create duplicate names in test

### PR DESCRIPTION
**Pull request by chsienki**
Trivial PR to make sure we don't generate duplicate names in the `AddedSources_Are_Deterministic ` test, as seen in the failure case here: https://dev.azure.com/dnceng/public/_build/results?buildId=912502&view=ms.vss-test-web.build-test-results-tab&runId=29058568&resultId=110775&paneView=debug

### CODE REVIEW

**Comment by chsienki** [comment]
@dotnet/roslyn-compiler for review please.

File: src/Compilers/CSharp/Test/Semantic/SourceGeneration/AdditionalSourcesCollectionTests.cs [hunk 1]
```diff
@@ -106,7 +106,15 @@ public void AddedSources_Are_Deterministic()
             asc = new AdditionalSourcesCollection(".cs");
             for (int i = 0; i < 1000; i++)
             {
-                names[i] = r.NextDouble().ToString() + ".cs";
+                // ensure the name is unique (we've seen CI failures where we generate duplicate names)
+                var name = "";
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Semantic/SourceGeneration/AdditionalSourcesCollectionTests.cs:110]: Could we just use a GUID?

Comment (chsienki) [reply src/Compilers/CSharp/Test/Semantic/SourceGeneration/AdditionalSourcesCollectionTests.cs:110]: Yes, that would work too. Although, you known, I never thought i'd see us get two random values duplicate in 1000 iterations either, but here we are.

Comment (cston) [reply src/Compilers/CSharp/Test/Semantic/SourceGeneration/AdditionalSourcesCollectionTests.cs:110]: Consider using `TestBase.GetUniqueName()`.

Comment (chsienki) [reply src/Compilers/CSharp/Test/Semantic/SourceGeneration/AdditionalSourcesCollectionTests.cs]: Hmm, TIL, thanks @cston

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Test/Semantic/SourceGeneration/AdditionalSourcesCollectionTests.cs (hunk 1 above)
Comment (333fred) [reply src/Compilers/CSharp/Test/Semantic/SourceGeneration/AdditionalSourcesCollectionTests.cs]: > Yes, that would work too. Although, you known, I never thought i'd see us get two random values duplicate in 1000 iterations either, but here we are.

If we see colliding GUIDs, we have bigger problems :).

**Review by 333fred — APPROVED** [review]

**Pull request merged**